### PR TITLE
CLIMATE-597 - Subsetting a dataset should update the names attribute

### PR DESCRIPTION
**Pull request by kwhitehall**
Made the naming of the subregion optional at it is not a necessity, but a nice feature to keep track of datasets.

### CODE REVIEW

File: ocw/dataset_processor.py [hunk 1]
```diff
@@ -173,6 +176,11 @@ def subset(subregion, target_dataset):
     # Get subregion indices into subregion data
     dataset_slices = _get_subregion_slice_indices(subregion, target_dataset)
 
+    if subregion_name:
```
Comment (MJJoyce) [inline ocw/dataset_processor.py]: Hey Kim,

You can simplify this if-else block down to simply

```
if not subregion_name:
    subregion_name = target_dataset.name
```

You'll need to update the line below where you pass the name as well if you change this of course =D

**Comment by MJJoyce** [comment]
Hey Kim,

Thanks for the pull request. Could you update the title so this links properly with the relevant JIRA ticket?

Also, this change seems like a great place for a test or two!

File: ocw/dataset_processor.py (hunk 1 above)
Comment (kwhitehall) [inline ocw/dataset_processor.py]: +1

**Comment by kwhitehall** [comment]
Hey @MJJoyce, I don't follow what you mean by a test... An example?

**Comment by MJJoyce** [comment]
Hey Kim,

So we currently test subsetting functionality in https://github.com/apache/climate/blob/master/ocw/tests/test_dataset_processor.py#L174

At the moment it looks like we don't have any tests to ensure that variables are correctly propagated through to the resulting dataset when we do subset. However, it would probably be wise to start. This seems like a great first step to getting that setup since we're adding new functionality here. I would say a sanity check test to make sure that the target dataset's name is propagated and a test that makes sure your new functionality works as expected would be awesome.

You could check the dataset tests at https://github.com/apache/climate/blob/master/ocw/tests/test_dataset.py#L25 for an example of what I'm talking about.

If you have any questions let me know!

**Comment by MJJoyce** [comment]
Oh! I just remembered. Since you're adding this functionality we should also update the safe_subset function so we can use it as well! This is probably a good pull request/ticket to do it under since they're pretty related I think. Thoughts?

https://github.com/apache/climate/blob/master/ocw/dataset_processor.py#L198

**Comment by kwhitehall** [comment]
w.r.t. to the safe_subset function, the logic is not affected and neither is the documentation that goes with that function so I think we are ok there.

**Comment by MJJoyce** [comment]
@kwhitehall Regarding the safe_subset function, it lacks the ability to use this added optional name argument. That seems like something worth having in my opinion. My recommendation wasn't to update the docs or anything on that, it was to add the same functionality for it under this ticket since they're so related.

Also, it looks like you got a few merge commits and stuff in there. A `git rebase -i master` and squashing those follow up commits would clean things up nicely!

**Pull request merged**